messaging/src/main/java/org/axonframework/messaging/deadletter/Cause.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2010-2022. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.messaging.deadletter;
+
+/**
+ * Contract describing the cause for {@link DeadLetter dead-lettering} a {@link org.axonframework.messaging.Message}.
+ * These objects typically reflects a {@link Throwable}.
+ *
+ * @author Steven van Beelen
+ * @author Mitchell Herrijgers
+ * @since 4.6.0
+ */
+public interface Cause {
```

Comment:
Should we add a "field" for details? I can imagine when the cause is an exception, that it's valuable having the stacktrace as well.

Reply:
Sounds reasonable...would that be a `Cause` object? 
The fact we don't use `Throwable` here is to simplify the de-/serialization that a, for example, JPA implementation needs to do. Trying to think how we would offer such a field :thinking: